### PR TITLE
Refactor MB4 related test cases

### PR DESCRIPTION
**Pull request by himasha**
## Purpose
1. Refactored all MB test related class packages from 'test.java.org.wso2.ei.mb.test.amqp' to 'org.wso2.ei.mb.test.amqp' to avoid unnecessary imports within a package.
2. Fixed formatting issues in amqp-tests.
3. Added a new test case to test basic user authentication scenario.

## Goals
Test the functionality of the newly introduced Authentication interface and its default implementation.

## Approach
N/A

## User stories
N/A

## Release note
N/A

## Documentation
N/A : Still an on-going development.
## Training
N/A : Still an on-going development.

## Certification
N/A : Still an on-going development.

## Marketing
N/A : Still an on-going development.

## Automation tests
 - Unit tests 
  N/A : Still an on-going development. Current implementation simply exposes a new interface for authentication implementation.
 - Integration tests
   > Added new test-case org.wso2.ei.mb.test.UserAuthTestCase

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes

## Samples
N/A : Still an on-going development.

## Related PRs
N/A 

## Migrations (if applicable)
N/A 

## Test environment
N/A : Still an on-going development.

## Learning
N/A

### CODE REVIEW

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/BrokerTest.java [hunk 1]
```diff
@@ -55,8 +56,8 @@ public void init() throws Exception {
             try {
                 String tempCarbonHome = serverManager.setupServerHome(archiveFilePath);
 
-                // load client configs to map
-                configurationReader = new ConfigurationReader();
+                // load client configs to map with valid username/password
+                configurationReader = new ConfigurationReader(false);
```
Comment (a5anka) [inline integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/BrokerTest.java]: Passing a boolean is not very readable. Shall we use a static factory method for this?

```java
configurationReader = ConfigurationReader.getClientConfigForInvalidUser();
```

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * @throws Exception
+     */
+    @Test(groups = "wso2.mb")
+    public void performInvalidConnectionTestCase() throws Exception {
+        QueueReceiver queueReceiver;
+        try {
+            queueReceiver = new QueueReceiver("InvalidConnectionQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                    invalidCredentialsConfig);
+            Assert.assertNull(queueReceiver, "Invalid user based connection made.");
+            Assert.fail("Expected exception not thrown for invalid user");
```
Comment (a5anka) [inline integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java]: This is cool. :+1:

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        } catch (JMSException e) {
```
Comment (a5anka) [inline integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java]: Shouldn't we always expect this exception? If yes can't we specify this in annotation?

```java
@Test(expectedExceptions = JMSException.class)
```

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/BrokerTest.java (hunk 1 above)
Comment (himasha) [reply integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/BrokerTest.java]: +1. Will update it.

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java (hunk 3 above)
Comment (himasha) [reply integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java]: Agree (y). Will update it .

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/BrokerTest.java (hunk 1 above)
Comment (himasha) [reply integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/BrokerTest.java]: Fixed with 2ded599.

File: integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java (hunk 3 above)
Comment (himasha) [reply integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/UserAuthTestCase.java]: Fixed with 2ded599

**Pull request merged**